easybuild/scripts/rpath_args.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+##
+# Copyright 2016-2016 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Utility script used by RPATH wrapper script;
+output is statements that define the following environment variables
+* $CMD_ARGS: new list of command line arguments to pass
+* $RPATH_ARGS: command line option to specify list of paths to RPATH
+
+author: Kenneth Hoste (HPC-UGent)
+"""
+import os
+import re
+import sys
+
+
+cmd = sys.argv[1]
+rpath_filter = sys.argv[2]
+args = sys.argv[3:]
+
+# wheter or not to use -Wl to pass options to the linker
+if cmd in ['ld', 'ld.gold']:
+    flag_prefix = ''
+else:
+    flag_prefix = '-Wl,'
+
+if rpath_filter:
+    rpath_filter = rpath_filter.split(',')
+else:
+    # don't RPATH system library locations by default
+    rpath_filter = ['/lib', '/usr/lib']
+
+version_mode = False
+cmd_args, cmd_args_rpath = [], []
+
+# process list of original command line arguments
+idx = 0
+while idx < len(args):
+
+    arg = args[idx]
+
+    # if command is run in 'version check' mode, make sure we don't include *any* -rpath arguments
+    if arg in ['-v', '-V', '--version', '-dumpversion']:
```

Comment:
No, we need to cover all possible `gcc` arguments for which we should not inject `-rpath` flags.

`gcc -v` should actually be called as `gcc -v`, not `gcc -Wl,-rpath=$ORIGIN/..lib ... -v`, because is breaks the command...